sonarts-core/tests/rules/noInMisuseRule/noInMisuseRule.lint.ts

```diff
@@ -0,0 +1,57 @@
+function nokFunc1() {
+  let arr = ["a", "b", "c"];
+
+  let expectedValue = "b";
+  if (expectedValue in arr) {
+//    ^^^^^^^^^^^^^^^^^^^^    {{Use "indexOf" or "includes" (available from ES2016) instead.}}
+    return expectedValue + " found in the array";
+  } else {
+    return expectedValue + " not found";
+  }
+}
+
+function nokFunc2() {
+  let arr = ["a", "b", "c"];
+
+  let expectedValue = "1"; // index #1 is corresponding to the value "b"
+  if (expectedValue in arr) {
+//    ^^^^^^^^^^^^^^^^^^^^    {{Use "indexOf" or "includes" (available from ES2016) instead.}}
+    return expectedValue + " found in the array";
+  } else {
+    return expectedValue + " not found";
+  }
+}
+
+
+function nokFunc3(a: any) {
+  const result = "car" in Object.keys(a) ? "something" : "someething else";
```

Comment:
@andrea-guarino-sonarsource what does this test testing? add comment or meaningful name to the func